### DISCUSSION TITLE
Roadmap and priorities suggestions for 1.0

### DISCUSSION THREAD

**Discussion by generalloki** [alternetsoft/AlternetUI#123] [Polls]
Here we collect suggestions and ideas for Version 1.0. 
What need to be done before release?

**Comment by generalloki** [comment]
@neoxeo suggested:

Priorities for me for version 1.0 :
- fix current known bugs
- if possible solve problem "need to compile" before the possibility to use new control created in uixml into the code (cs file)
- if possible, change size, height, width to use them into uixml to set values (not use bounds)
- if possible, add RichTextBox
- add two new samples : one for an existing simple winforms application converted to Alternet UI and same for a WPF existing application

**Comment by neoxeo** [comment]
Others suggestions:

Update to the last stable release of Webview2 ?

Add .Net8.0 with comment "not stable version" ?

Add tutorials, video or document : 
- how to add properties, methods, event not already exist in AlternetUI but present into WxWidgets
- how to create a new component using API
- update how to create a new application with ANUI from scratch
-

**Comment by generalloki** [comment]
Before 1.0 release, we need to implement the following controls:

- DONE. Advanced ToolBars (https://wiki.wxwidgets.org/WxAUI).
- MonthCalendar ([wxCalendarCtrl](https://docs.wxwidgets.org/3.0/classwx_calendar_ctrl.html)).
- WORKING ON. PropertyGrid (100% needed, useful for control testing).([wxPropertyGrid](https://docs.wxwidgets.org/3.0/classwx_property_grid.html))
- NOT POSSIBLE (Button implementations are bad on Macos). SpeedButton/Label with Image (Button descendants).

Not sure, but possibly:
- HelpProvider ([wxHtmlHelpController](https://docs.wxwidgets.org/3.0/classwx_html_help_controller.html)) 
- RichTextBox  ([wxRichTextBox](https://docs.wxwidgets.org/3.0/classwx_rich_text_box.html)) 
- [wxHtmlListBox](https://docs.wxwidgets.org/3.0/classwx_html_list_box.html)
- [wxVListBox](https://docs.wxwidgets.org/3.0/classwx_v_list_box.html)
- [wxSimpleHtmlListBox](https://docs.wxwidgets.org/3.0/classwx_simple_html_list_box.html)
- [wxEditableListBox](https://docs.wxwidgets.org/3.0/classwx_editable_list_box.html)
- wxDataViewCtrl 
- [wxScrollBar](https://docs.wxwidgets.org/3.0/classwx_scroll_bar.html)
- FileListView and FileTreeView.

**Comment by generalloki** [comment]
We need some demo explorer with list of demos and descriptions like in ALternet Studio. 
![alternetsampledemorun](https://github.com/alternetsoft/AlternetUI/assets/59140005/78455094-dc47-4f90-afe9-b4f85b29ca80)

**Comment by generalloki** [comment]
- We need PictureBox control demo in ControlSample.
- Also check that WinForms features are implemented. 
- Also image related features from DrawContext should be implemented.

**Comment by neoxeo** [comment]
What ? I was sure that PictureBox was already present.
For sure, it is necessary! +1

**Comment by generalloki** [comment]
PictureBox is implemeted on some level but we need to have demo and to add some features.

**Comment by neoxeo** [comment]
Yes after reflection I remember that Employees Sample have one.
Ok, agree with you to have a demo to show features.
Thanks for your answer.

**Comment by generalloki** [comment]
Add Ribbon user interface
https://docs.wxwidgets.org/3.0/group__group__class__ribbon.html
![ribbon](https://github.com/alternetsoft/AlternetUI/assets/59140005/f0e0f5ac-6137-447e-b2aa-6d1135a4f4b9)

**Comment by generalloki** [comment]
We need to add SQLite database support.
This is probably not for 1.0 but for later releases.

**Comment by generalloki** [comment]
We need to support wxCursor set/get in controls
